### DISCUSSION TITLE
Date Picker - validation issues

### DISCUSSION THREAD

**Discussion by mixmaniax** [Budibase/budibase#8402] [Help] [answered]
I have a problem with date validation on a form.

Scenario:
I have a holiday request form including a "from" and "to" date picker. Now there are some days which should be blocked (not allowed) because all staff need to be present at work. I would like validation to block the form submit if a blocked day is spresent whitin the selected date range "from" and "to".

I built a javascript which correctly checks both date fields against some blocked dates. The js returns 'true' (if a blocked date was found) or 'false' if the dates requested are valid. For the moment I implemented this js to run under a "Show Message" action of the form submit button to see the result. Obviously I need a way to show a validation error if the JS returns 'true'. Any help is appreciated on how and I could achieve this.

Thanks a lot.

**Comment by melohagan** [comment]
Hey @mixmaniax 

You don't necessarily need JS for this.

1. Click on your date picker
2. Click `Configure validation` in the settings panel
![Screenshot 2022-10-26 at 16 13 03](https://user-images.githubusercontent.com/101575380/198065188-bbfb792f-3823-44e2-abee-10c44e764f37.png)
3. Add a rule **Must not equal** Value *INVALID_DATE_HERE*
![Screenshot 2022-10-26 at 16 13 57](https://user-images.githubusercontent.com/101575380/198065392-69adc6d1-685d-46b8-bfd8-eade38d6cd0c.png)
4. You can add a rule for as many invalid dates as you like, and can use *Binding* instead of *Value* if the date is being pulled dynamically from somewhere.

![Screenshot 2022-10-26 at 16 17 16](https://user-images.githubusercontent.com/101575380/198066200-86fa9bf9-1f22-44e8-acae-0871bc0349a6.png)

**Reply by mixmaniax** [reply]
Hi @melohagan 

Thank you, this won't work in my use case i guess.

I have lets say two date picker.
Date picker FROM and date picker TO.
The user requests holiday from October 1st 2022 to October 30, 2022. Unfortunately October 7th, 2022 is a blocked day. So the within the range selected by the user there is a blocking date, so that should raise a validation error...

Any idea how to solve this, with or without JS?

Thank you again.

**Reply by melohagan** [reply]
Oh I understand, apologies! I'll have to have a think about that one.

**Comment by melohagan** [comment, marked as answer]
Hey @mixmaniax 

I had a play around with [Conditional UI](https://docs.budibase.com/docs/conditions) and using Data Provider filters to determine if the blocked dates are within the range.
Unfortunately it was a bit janky, and you could potentially quickly click save while the data provider was updating with newly selected dates.

The real solution is to add a setting within the date picker to allow for specific dates to be disabled. Then you won't need to worry about validation at all.
I've raised an issue here: https://github.com/Budibase/budibase/issues/8415

**Reply by mixmaniax** [reply]
@melohagan 
Thanks a lot - hope this feature gets implemented

**Reply by tokkisva** [reply]
I am also kind of stuck in a similar situation.

I'm trying to create a validation that requires the date picked to be within the current week.  

I'm thinking this can only be done in javascript, and I do have some code that can check to see whether a date is within 'this week' but I'm not exactly sure how to implement it or if it is even possible within BB right now. 

@melohagan do you know if BB could do something like the above at this time? If I need to open a new thread on this please let me know.

Thanks!
